src/fparser/splitline.py

```diff
@@ -176,313 +182,122 @@ def splitquote(line, stopchar=None, lower=False, quotechars = '"\''):
-            try:
-                char = fifo_pop()
-            except IndexError:
-                return String(''.join(l))
-        # else continued string
-        while 1:
-            if char==quotechar and not nofslashes % 2:
-                l_append(char)
-                self.quotechar = None
-                break
-            if char=='\\':
-                nofslashes += 1
-            else:
-                nofslashes = 0
-            l_append(char)
-            try:
-                char = fifo_pop()
-            except IndexError:
-                break
-        return String(''.join(l))
 
-def splitparen(line,paren='()'):
+def splitparen(line, paren_open="([", paren_close=")]"):
     """
-    Fast LineSplitterParen.
+    Splits a line into top-level parenthesis and not-parenthesised
+    parts. E.g.: "a( (1+2)*3) = b(x)" becomes:
+    ["a", "( (1+2)*3)", " = b", "(x)"]
+    :param str line: the string to split.
+    :param str paren_open: The characters that define an open parentheses.
+    :param str paren_close: The characters that define a closing parentheses.
+    The paren_open and paren_close strings must be matched in order:
+    paren_open[x] is closed by paren_close[x].
     """
-    stopchar = None
-    startchar, endchar = paren[0],paren[1]
-
-    items = []
-    i = 0
-    while 1:
-        try:
-            char = line[i]; i += 1
-        except IndexError:
-            break
-        nofslashes = 0
-        charlist = []
-        if stopchar is None:
-            # search for parenthesis start
-            while 1:
-                if char==startchar and not nofslashes % 2:
-                    stopchar = endchar
-                    i -= 1
-                    break
-                if char=='\\':
-                    nofslashes += 1
-                else:
-                    nofslashes = 0
-                charlist.append(char)
-                try:
-                    char = line[i]; i += 1
-                except IndexError:
-                    break
-            item = ''.join(charlist)
-        else:
-            nofstarts = 0
-            while 1:
-                if char==stopchar and not nofslashes % 2 and nofstarts==1:
-                    charlist.append(char)
-                    stopchar = None
-                    break
 
-                if char=='\\':
-                    nofslashes += 1
-                else:
-                    nofslashes = 0
+    assert len(paren_open) == len(paren_close)
+
+    items = []   # Result list
+    i = 0        # Index of current character
+    num_backslashes = 0   # Counts consecutivre "\" characters
+    # Empty if outside quotes, or set to the starting (and therefore
+    # also the ending) quote character while reading text inside quotes.
+    inside_quotes_char = ""
+    start = 0    # Index of start of current part.
+    stack = []   # Stack keeping track of required closing brackets
+
+    i = -1
+    for i, char in enumerate(line):
```

Comment:
Please could you change `i` to e.g. `idx` (pep8)

Reply:
Neither pep8 nor pylint complained about this, and I personally prefer using i as loop index - but I fixed it.